docs/code-search/code-navigation/auto_indexing.mdx

```diff
@@ -0,0 +1,320 @@
+
+<Callout type="note">This feature is in beta for self-hosted customers.</Callout>
+
+With Sourcegraph deployments supporting [executors](/admin/executors/), your repository contents can be automatically analyzed to produce a code graph index file. Once [auto-indexing is enabled](/code_navigation/how-to/enable_auto_indexing) and [auto-indexing policies are configured](/code_navigation/how-to/configure_auto_indexing), repositories will be periodically cloned into an executor sandbox, analyzed, and the resulting index file will be uploaded back to the Sourcegraph instance.
+
+## Enable Code Navigation auto-indexing
+
+<Callout type="note">This feature is in beta for self-hosted customers.</Callout>
+
+The following docs explains how to turn on [auto-indexing](/code_navigation/explanations/auto_indexing) on your Sourcegraph instance to enable [precise code navigation](/code_navigation/explanations/precise_code_navigation).
+
+### Deploy executors
+
+First, [deploy the executor service](/admin/executors/deploy_executors) targeting your Sourcegraph instance. This will provide the necessary compute resources that clone the target Git repository, securely analyze the code to produce a code graph data index, then upload that index to your Sourcegraph instance for processing.
+
+### Enable index job scheduling
+
+Next, enable the precise code navigation auto-indexing feature by enabling the following feature flag in your Sourcegraph instance's site configuration.
+
+```yaml
+{
+  "codeIntelAutoIndexing.enabled": true
+}
+```
+
+This step will control the scheduling of indexing jobs which are made available to the executors deployed in the previous step.
+
+### Configure auto-indexing policies
+
+Once auto-indexing has been enabled, [create auto-indexing policies](/code_navigation/how-to/configure_auto_indexing) to control the set of repositories and commits that are eligible for indexing. Note that policies only select repositories and branches/tags/commits that can be indexed, not how they are indexed.
+
+### Configure auto-indexing jobs
+
+For repository and commit pairs that are marked as eligible for indexing, the index job is either [inferred from the project](/code_navigation/explanations/auto_indexing_inference), or [explicitly configured](/code_navigation/how-to/configure_auto_indexing#explicit-index-job-configuration) on a per-repository basis.
+
+### Tune the index scheduler
+
+The frequency of index job scheduling can be tuned via the following environment variables read by `worker` service containers running the [`codeintel-auto-indexing`](/admin/workers#codeintel-auto-indexing) task.
+
+- **`PRECISE_CODE_INTEL_AUTO_INDEXING_TASK_INTERVAL`**: The frequency with which to run periodic codeintel auto-indexing tasks. Default is every 2 minutes
+- **`PRECISE_CODE_INTEL_AUTO_INDEXING_REPOSITORY_PROCESS_DELAY`**: The minimum frequency that the same repository can be considered for auto-index scheduling. Default is every 24 hours
```

Comment:
The wording here is confusing because the variables correspond to time durations, and the description says "frequency" which is technically the reciprocal of the time duration between consecutive iterations.